packages/status-js/src/client/client.ts

```diff
@@ -18,14 +18,32 @@ export interface ClientOptions {
 class Client {
   public waku: Waku
   public readonly wakuMessages: Set<string>
+  /**
+   * Tracks open connections which had their streams silently destroyed
+   * and closes them so new connections can be automatically created.
+   *
+   * For example, in case of a websocket in browser which did not
+   * have its close event emitted.
+   *
+   * Note: Detection of the stream removal depends on active (by user)
+   * or pasive (with `Waku.relayKeepAlive`) message sending.
+   *
+   * Note: This is only a workaround (@see https://github.com/libp2p/js-libp2p/issues/939).
+   */
+  #wakuDisconnectionTimer: ReturnType<typeof setInterval>
```

Comment:
Cc @D4nte

Reply:
Do you think this would help? https://github.com/libp2p/js-libp2p/pull/1221

Let's see once I finish the upgrade. We can also do a PR upstream if we know how to fix it.

Reply:
>Do you think this would help? https://github.com/libp2p/js-libp2p/pull/1221

By the sound of it, I'm afraid not. In that case, it seams like when a connection "ends" the stream doesn't. While in our case, when a stream, or even a socket, ends it is the connection that doesn't end nor recreates it.

The chain of events is backwards, I believe.

> Let's see once I finish the upgrade.
>
> [– https://github.com/status-im/js-waku/pull/803]

I agree. Great job on that one, looking forward 💪.

Reply:
Also pulled into https://github.com/xmtp/xmtp-js/pull/128.